cmd/cli/handler_client.go

```diff
@@ -101,6 +101,13 @@ func (h *ClientHandler) CreateClient(cmd *cobra.Command, args []string) {
 	ek, encryptSecret, err := newEncryptionKey(cmd, nil)
 	cmdx.Must(err, "Failed to load encryption key: %s", err)
 
+	var metadata models.JSONRawMessage
+	metadataStr := flagx.MustGetString(cmd, "metadata")
+	if metadataStr != "" {
+		err = json.Unmarshal([]byte(flagx.MustGetString(cmd, "metadata")), &metadata)
+		cmdx.Must(err, "Failed to parse metadata: %s", err)
```

Comment:
Can you please use https://github.com/tidwall/gjson/blob/master/gjson.go#L2549 instead and improve the error message a bit? :)

Reply:
Sure :) Although the JSON string, will it still need to also be parsed out to type `JSONRawMessage` to pass into the `OAuth2Client` struct?

Reply:
Yes, but `JSONRawMessage` is just `[]byte()`, so:

```
var JSONRawMessage = []byte(`{"foo":"bar"}`)
```

Reply:
So if I set `Metadata` as above as just a raw string from the CLI input, then it gets added to the client as: 
```json
"metadata": "{\"foo\":\"bar\"}",
```
Which is not what I would expect to see. However if I do something like:
```go
var metadata interface{}
json.Unmarshal([]byte(flagx.MustGetString(cmd, "metadata")), &metadata)
```

Then in the resulting client output I get:
```json
"metadata": {
	"foo": "bar"
},
```

Which is much more like what I was expecting

Reply:
```
var metadata json.RawMessage = []byte(flagx.MustGetString(cmd, "metadata"))
```

is equivalent and to much faster than

```
var metadata interface{}
json.Unmarshal([]byte(flagx.MustGetString(cmd, "metadata")), &metadata)
```

:)

Reply:
Ohhhh, my bad, yeah I was missing the fact `JSONRawMessage` implemented an Unmarshaler! 

I'll update that now, thanks!

Reply:
awesome :)

Reply:
Apologies, I'd made some local changes and then completely forgot about this PR 😬 I've just pushed now, does that look more like what you'd expect? It seemed long winded but `gjson.Valid` returned a bool so I wasn't sure the preferred way to error out, at the moment it's just printing there was an error and carrying on anyway